### PR TITLE
Allow raters to nominate contributors during the rating period.

### PR DESCRIPTION
**Pull request by noahsaso**
In case contributors miss the submission period, DAO members should be able to nominate contributors while ratings are open.

Only the rater who nominated a contribution can update that contribution, and a rater cannot override a contribution that already exists.

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -113,6 +114,19 @@ Submit a contribution to the active survey. Anyone can do this while contributio
...
 
+### `POST /:dao/nominate`
+
+Nominate a contribution to the active survey. Any DAO member can do this while ratings are being accepted.
+
+#### Request
+
+```ts
+{
+  contributor: string
```
Comment (elsehow) [inline DESIGN.md:127]: Is this an address? A handle? Ideally we'd have both...

File: src/index.ts [hunk 2]
```diff
@@ -72,6 +73,14 @@ router.post(
   loadActiveSurveyForDao,
   submitContribution
 )
+// Nominate contribution.
+router.post(
+  '/:dao/nominate',
+  loadDaoFromParams,
+  loadActiveSurveyForDao,
+  authDaoMemberAtSurveyCreationBlockHeightMiddleware,
```
Comment (elsehow) [inline src/index.ts:81]: Incredible name, 10/10. Long, but 0 ambiguity about what this is and what it does.

**Review by elsehow — APPROVED** [review]
This looks great. Begged the question for me: are contributors defined by their human-readable name, their address, or both? Seems relevant to the drop to have the address, and to the rating to have the human readable name... now, ideally, that would all be handled by the profile system, so I'm assuming `contributor` is *always* an address. If that's the case, where do we document that? Where do we validate it? Maybe not in the worker, but if not there, where?

These questions fall outside the scope of this PR, so this is a definite approve for now.

File: DESIGN.md (hunk 1 above)
Comment (noahsaso) [reply DESIGN.md:127]: This is a public key. In line with all the other identity fields (`rater` and `contributor`), entities are only represented by their public keys. We could add `PublicKey` suffixes to all of them, but that might be unnecessarily verbose, and I kinda like the 1:1 mapping of "user is not different from public key".

**Comment by noahsaso** [comment]
> This looks great. Begged the question for me: are contributors defined by their human-readable name, their address, or both? Seems relevant to the drop to have the address, and to the rating to have the human readable name... now, ideally, that would all be handled by the profile system, so I'm assuming `contributor` is _always_ an address. If that's the case, where do we document that? Where do we validate it? Maybe not in the worker, but if not there, where?
> 
> These questions fall outside the scope of this PR, so this is a definite approve for now.

no addresses, only public keys! We should definitely be validating that public keys are indeed public keys. Great point, will make an issue.

**Comment by noahsaso** [comment]
> > This looks great. Begged the question for me: are contributors defined by their human-readable name, their address, or both? Seems relevant to the drop to have the address, and to the rating to have the human readable name... now, ideally, that would all be handled by the profile system, so I'm assuming `contributor` is _always_ an address. If that's the case, where do we document that? Where do we validate it? Maybe not in the worker, but if not there, where?
> > These questions fall outside the scope of this PR, so this is a definite approve for now.
> 
> no addresses, only public keys! We should definitely be validating that public keys are indeed public keys. Great point, will make an issue.

Just kidding I can't make an issue on a fork. Damn.

**Comment by noahsaso** [comment]
Added validation.

**Pull request merged**